### ISSUE TITLE
[ci][sdk/nodejs] `TestLanguage/l2-resource-simple` now failing

### ISSUE THREAD

**Issue by dixler** [pulumi/pulumi#14781]
### What happened?

Seeing failures in [CI / Unit Test / sdk/nodejs/cmd/pulumi-language-nodejs on ubuntu-latest/current](https://github.com/pulumi/pulumi/actions/runs/7131708694/job/19421446621#logs)

I suspect either:
- a dependency wasn't pinned.
- https://github.com/pulumi/pulumi/pull/14415

https://github.com/pulumi/pulumi/actions/runs/7131798042/job/19421319413
https://github.com/pulumi/pulumi/actions/runs/7131708694/job/19421446621

### Example

```
Error: node_modules/@pulumi/pulumi/proto/alias_pb.d.ts(7,23): error TS7016: Could not find a declaration file for module 'google-protobuf'. '/tmp/TestLanguage3261282042/002/sdks/simple-1.0.0/node_modules/google-protobuf/google-protobuf.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/google-protobuf` if it exists or add a new declaration (.d.ts) file containing `declare module 'google-protobuf';`
Error: node_modules/@pulumi/pulumi/proto/engine_grpc_pb.d.ts(9,43): error TS7016: Could not find a declaration file for module 'google-protobuf/google/protobuf/empty_pb'. '/tmp/TestLanguage3261282042/002/sdks/simple-1.0.0/node_modules/google-protobuf/google/protobuf/empty_pb.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/google-protobuf` if it exists or add a new declaration (.d.ts) file containing `declare module 'google-protobuf/google/protobuf/empty_pb';`
Error: node_modules/@pulumi/pulumi/proto/engine_pb.d.ts(7,23): error TS7016: Could not find a declaration file for module 'google-protobuf'. '/tmp/TestLanguage3261282042/002/sdks/simple-1.0.0/node_modules/google-protobuf/google-protobuf.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/google-protobuf` if it exists or add a new declaration (.d.ts) file containing `declare module 'google-protobuf';`
Error: node_modules/@pulumi/pulumi/proto/engine_pb.d.ts(8,43): error TS7016: Could not find a declaration file for module 'google-protobuf/google/protobuf/empty_pb'. '/tmp/TestLanguage3261282042/002/sdks/simple-1.0.0/node_modules/google-protobuf/google/protobuf/empty_pb.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/google-protobuf` if it exists or add a new declaration (.d.ts) file containing `declare module 'google-protobuf/google/protobuf/empty_pb';`
Error: node_modules/@pulumi/pulumi/proto/plugin_pb.d.ts(7,23): error TS7016: Could not find a declaration file for module 'google-protobuf'. '/tmp/TestLanguage3261282042/002/sdks/simple-1.0.0/node_modules/google-protobuf/google-protobuf.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/google-protobuf` if it exists or add a new declaration (.d.ts) file containing `declare module 'google-protobuf';`
Error: node_modules/@pulumi/pulumi/proto/provider_pb.d.ts(7,23): error TS7016: Could not find a declaration file for module 'google-protobuf'. '/tmp/TestLanguage3261282042/002/sdks/simple-1.0.0/node_modules/google-protobuf/google-protobuf.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/google-protobuf` if it exists or add a new declaration (.d.ts) file containing `declare module 'google-protobuf';`
Error: node_modules/@pulumi/pulumi/proto/provider_pb.d.ts(9,43): error TS7016: Could not find a declaration file for module 'google-protobuf/google/protobuf/empty_pb'. '/tmp/TestLanguage3261282042/002/sdks/simple-1.0.0/node_modules/google-protobuf/google/protobuf/empty_pb.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/google-protobuf` if it exists or add a new declaration (.d.ts) file containing `declare module 'google-protobuf/google/protobuf/empty_pb';`
Error: node_modules/@pulumi/pulumi/proto/provider_pb.d.ts(10,44): error TS7016: Could not find a declaration file for module 'google-protobuf/google/protobuf/struct_pb'. '/tmp/TestLanguage3261282042/002/sdks/simple-1.0.0/node_modules/google-protobuf/google/protobuf/struct_pb.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/google-protobuf` if it exists or add a new declaration (.d.ts) file containing `declare module 'google-protobuf/google/protobuf/struct_pb';`
Error: node_modules/@pulumi/pulumi/proto/resource_grpc_pb.d.ts(9,43): error TS7016: Could not find a declaration file for module 'google-protobuf/google/protobuf/empty_pb'. '/tmp/TestLanguage3261282042/002/sdks/simple-1.0.0/node_modules/google-protobuf/google/protobuf/empty_pb.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/google-protobuf` if it exists or add a new declaration (.d.ts) file containing `declare module 'google-protobuf/google/protobuf/empty_pb';`
Error: node_modules/@pulumi/pulumi/proto/resource_grpc_pb.d.ts(10,44): error TS7016: Could not find a declaration file for module 'google-protobuf/google/protobuf/struct_pb'. '/tmp/TestLanguage3261282042/002/sdks/simple-1.0.0/node_modules/google-protobuf/google/protobuf/struct_pb.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/google-protobuf` if it exists or add a new declaration (.d.ts) file containing `declare module 'google-protobuf/google/protobuf/struct_pb';`
Error: node_modules/@pulumi/pulumi/proto/resource_pb.d.ts(7,23): error TS7016: Could not find a declaration file for module 'google-protobuf'. '/tmp/TestLanguage3261282042/002/sdks/simple-1.0.0/node_modules/google-protobuf/google-protobuf.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/google-protobuf` if it exists or add a new declaration (.d.ts) file containing `declare module 'google-protobuf';`
Error: node_modules/@pulumi/pulumi/proto/resource_pb.d.ts(8,43): error TS7016: Could not find a declaration file for module 'google-protobuf/google/protobuf/empty_pb'. '/tmp/TestLanguage3261282042/002/sdks/simple-1.0.0/node_modules/google-protobuf/google/protobuf/empty_pb.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/google-protobuf` if it exists or add a new declaration (.d.ts) file containing `declare module 'google-protobuf/google/protobuf/empty_pb';`
Error: node_modules/@pulumi/pulumi/proto/resource_pb.d.ts(9,44): error TS7016: Could not find a declaration file for module 'google-protobuf/google/protobuf/struct_pb'. '/tmp/TestLanguage3261282042/002/sdks/simple-1.0.0/node_modules/google-protobuf/google/protobuf/struct_pb.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/google-protobuf` if it exists or add a new declaration (.d.ts) file containing `declare module 'google-protobuf/google/protobuf/struct_pb';`
Error: node_modules/@pulumi/pulumi/proto/source_pb.d.ts(7,23): error TS7016: Could not find a declaration file for module 'google-protobuf'. '/tmp/TestLanguage3261282042/002/sdks/simple-1.0.0/node_modules/google-protobuf/google-protobuf.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/google-protobuf` if it exists or add a new declaration (.d.ts) file containing `declare module 'google-protobuf';`
    language_test.go:135: 
        	Error Trace:	/home/runner/work/pulumi/pulumi/sdk/nodejs/cmd/pulumi-language-nodejs/language_test.go:135
        	Error:      	Received unexpected error:
        	            	rpc error: code = Unknown desc = sdk packing for simple: npm run build: exit status 2
        	Test:       	TestLanguage/l2-resource-simple
```

### Output of `pulumi about`

n/a

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by justinvp** [comment]
Can we skip the test now to unblock CI, and include a TODO comment that links to this issue to fix for real? Can you add the skip, @dixler?

**Comment by blampe** [comment]
This is breaking in provider builds as well, see for example https://github.com/pulumi/pulumi-cloudflare/actions/runs/7132092244/job/19422014451?pr=580

I think I have a fix in #14783.

**Comment by rquitales** [comment]
Also pulumi-kubernetes: https://github.com/pulumi/pulumi-kubernetes/actions/runs/7134619519/job/19429926959?pr=2694

Thanks for the fix!
